src/libDirectoryService/DSBlockPostProcessing.cpp

```diff
@@ -233,38 +229,48 @@ void DirectoryService::UpdateMyDSModeAndConsensusId() {
                                             .GetHashForRandom()
                                             .asBytes());
   }
-  // Check if I am the oldest backup DS (I will no longer be part of the DS
-  // committee)
-  if ((uint32_t)(m_consensusMyID + numOfIncomingDs) >=
-      m_mediator.m_DSCommittee->size()) {
+
+  // Find my new consensus ID.
+  DequeOfNode::iterator it;
+  bool isDropout = true;
+  for (it = m_mediator.m_DSCommittee->begin();
+       it != m_mediator.m_DSCommittee->end(); ++it) {
+    // Look for my public key.
+    if (m_mediator.m_selfKey.second == it->first) {
+      m_consensusMyID = it - m_mediator.m_DSCommittee->begin();
```

Comment:
better use the `distance` func

Reply:
Thanks, resolved in https://github.com/Zilliqa/Zilliqa/pull/1587/commits/95906a7d75a2be4474f3fd02b09ad55977828c08.